airflow/utils/helpers.py

```diff
@@ -205,10 +205,16 @@ def kill_descendant_processes(logger, pids_to_kill=None):
     logger.warn("Terminating descendant processes of {} PID: {}"
                 .format(this_process.cmdline(),
                         this_process.pid))
-    for descendant in descendant_processes:
+
+    temp_processes = descendant_processes
```

Comment:
Does this really take a copy?! Shouldn't it be something like this:
temp_processes = descendant_processes[:]